azure-mgmt-cdn/src/main/java/com/microsoft/azure/management/cdn/CdnEndpoint.java

```diff
@@ -111,25 +115,91 @@
     void start();
 
     /**
+     * Starts current stopped CDN endpoint asynchronously.
+     *
+     * @return a completable indicates completion or exception of the request
+     */
+    Completable startAsync();
+
+    /**
+     * Starts current stopped CDN endpoint asynchronously.
+     *
+     * @param callback the callback to call on success or failure
+     * @return a handle to cancel the request
+     */
+    ServiceFuture<Void> startAsync(ServiceCallback<Void> callback);
+
+    /**
      * Stops current running CDN endpoint.
      */
     void stop();
 
     /**
+     * Stops current running CDN endpoint asynchronously.
+     *
+     * @return a completable indicates completion or exception of the request
+     */
+    Completable stopAsync();
+
+
+    /**
+     * Stops current running CDN endpoint asynchronously.
+     *
+     * @param callback the callback to call on success or failure
+     * @return a handle to cancel the request
+     */
+    ServiceFuture<Void> stopAsync(ServiceCallback<Void> callback);
+
+    /**
      * Forcibly purges current CDN endpoint content.
      *
      * @param contentPaths the path to the content to be purged. Can describe a file path or a wild card directory.
      */
     void purgeContent(List<String> contentPaths);
 
     /**
+     * Forcibly purges current CDN endpoint content asynchronously.
+     *
+     * @param contentPaths the path to the content to be purged. Can describe a file path or a wild card directory.
+     * @return a completable indicates completion or exception of the request
```

Comment:
the `@return` comments - how about a standard language like "@return a representation of the deferred computation of this call"